Description: Do not fail the /thread/thread4 test if prctrl() fails. This happens on the Debian buildds.
Author: Martin Pitt <martin.pitt@ubuntu.com>
Forwarded: No, Debian buildd specific

--- a/glib/tests/thread.c
+++ b/glib/tests/thread.c
@@ -141,9 +141,11 @@
 
   getrlimit (RLIMIT_NPROC, &nl);
   nl.rlim_cur = 1;
-
   if ((ret = prlimit (getpid(), RLIMIT_NPROC, &nl, &ol)) != 0)
-    g_error ("prlimit failed: %s\n", g_strerror (ret));
+    {
+      g_debug ("prlimit failed: %s\n", g_strerror (ret));
+      return;
+    }
 
   error = NULL;
   thread = g_thread_try_new ("a", thread1_func, NULL, &error);
